rcldotnet/RCLdotnet.cs

```diff
@@ -199,8 +201,14 @@ public class RCLdotnet {
     public static Node CreateNode (string nodeName, string nodeNamespace) {
       IntPtr nodeHandle = IntPtr.Zero;
       int ret = RCLdotnetDelegates.native_rcl_create_node_handle (ref nodeHandle, nodeName, nodeNamespace);
-      Node node = new Node (nodeHandle);
-      return node;
+
+      if (ret != NODE_NAME_ERROR)
+      {
+        Node node = new Node (nodeHandle);
+        return node;
+      } else {
+        throw new InvalidNodeNameException(nodeName);
```

Comment:
This exception should either of these two in order to provide meaningful information to the caller:
- be split in several cases, one for each type of error
- or store the error code

Additionally, it should store the string returned by `rmw_get_error_string`

Reply:
I split this code in cases, and also I stored the result of `rmw_get_error_string` in the exception, although I am not sure that the way I have done it is correct.